tests/runfiles/fs_test.go

```diff
@@ -12,59 +12,185 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build go1.16
-// +build go1.16
-
 package runfiles_test
 
 import (
 	"io"
 	"io/fs"
 	"os"
+	"path"
 	"path/filepath"
 	"runtime"
+	"strings"
 	"testing"
-	"testing/fstest"
 
 	"github.com/bazelbuild/rules_go/go/runfiles"
+	"github.com/bazelbuild/rules_go/go/tests/runfiles/testfs"
 )
 
-func TestFS(t *testing.T) {
-	fsys, err := runfiles.New()
+var isBzlmodEnabled bool
+
+func init() {
```

Comment:
I think even just a simple `fs.WalkDir` test that verifies all dirs and files inside a small runfiles tree are visited would be sufficient. I can then point folks to the test as an example of how to use the API. It would also help document the expected tree traversal order and contents, which we could compare against future rules_go/bazel changes.

Reply:
I added such a test and it unearthed that walking source directories in runfiles behaved differently between manifest and directory. This was the case because `fs.WalkDir` doesn't descend into directory symlinks, which are used in this case by the directory-based but not the manifest-based runfiles. We could fix this either way, I went with resolving symlinks here. Let me know what you think.

Reply:
I would imagine supporting symlink traversal would be more useful here. Though Im not sure if there could be any recursive setup that screws with Go's `fs.WalkDir` assumption. Perhaps if that ever happened, then the caller could implement tracking in their WalkFunc and return SkipDirErr when a seen dir is revisited.

So the current approach sgtm.